packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -212,25 +213,37 @@ const DefaultSummaryConfiguration: ISummaryConfiguration = {
     maxAckWaitTime: 120000,
 };
 
-/**
- * Options for container runtime.
- */
-export interface IContainerRuntimeOptions {
-    // Flag that will generate summaries if connected to a service that supports them.
-    // This defaults to true and must be explicitly set to false to disable.
-    generateSummaries?: boolean;
-
-    // Delay before first attempt to spawn summarizing container
-    initialSummarizerDelayMs?: number;
-
-    // Flag that will disable garbage collection if set to true.
+export interface IGCRuntimeOptions {
+    /* Flag that will disable garbage collection if set to true. */
     disableGC?: boolean;
 
-    // Flag that will bypass optimizations and generate GC data for all nodes irrespective of whether the node
-    // changed or not.
+    /**
+     * Flag representing the summary's preference for enabling garbage collection.
+     * This is stored in the summary and unchangeable (for now). So this runtime option
+     * only takes affect on new containers.
+     * Currently if this is set to false, it will take priority and any container will
+     * never run GC.
+     */
+    summaryEnableGC?: boolean;
```

Comment:
Based on out conversation around what this flag means, we might consider naming it something that indicates this container is GC-ready. The other flag controls whether GC is enabled or not. Maybe rename this to something like `isReadyForGC` or `canRunGC`?

Reply:
What about `summaryAllowGC`?

Reply:
Changed to `gcAllowed`.